apps/api/src/app/notification-template/e2e/create-notification-templates.e2e.ts

```diff
@@ -297,6 +299,92 @@ describe('Create Notification template - /notification-templates (POST)', async
     expect(steps[0]._id).to.equal(steps[1]._parentId);
   });
 
+  it('should use sender name in email template', async function () {
+    const testTemplate: Partial<CreateNotificationTemplateRequestDto> = {
+      name: 'test email template',
+      description: 'This is a test description',
+      tags: ['test-tag'],
+      notificationGroupId: session.notificationGroups[0]._id,
+      steps: [
+        {
+          template: {
+            name: 'Message Name',
+            subject: 'Test email subject',
+            preheader: 'Test email preheader',
+            senderName: 'test',
+            content: [{ type: EmailBlockTypeEnum.TEXT, content: 'This is a sample text block' }],
+            type: StepTypeEnum.EMAIL,
+          },
+          filters: [],
+        },
+      ],
+    };
+
+    const { body } = await session.testAgent.post(`/v1/notification-templates`).send(testTemplate);
+
+    expect(body.data).to.be.ok;
+    const template: INotificationTemplate = body.data;
+
+    expect(template._notificationGroupId).to.equal(testTemplate.notificationGroupId);
+    const message = template.steps[0];
+    expect(message.template?.senderName).to.equal('test');
+  });
+
+  it('should build factory integration', () => {
```

Comment:
Maybe this unit test should live closer to the SendMessageEmail use case and not in this E2E test suite? Like creating a `send-message-email.usecase.spec.ts` file, even if will only hold one test suite so far (for the `buildFactoryIntegration` method) but we might split this test in 3 so the expectations get documented.

Reply:
I kind of needing it to be in e2e so I can use the testserver to get an instance of the SendMessageEmail class